torch/csrc/jit/ir/constants.cpp

```diff
@@ -161,6 +161,11 @@ c10::optional<IValue> toIValue(const Value* v) {
     const auto& list = node->ival(attr::value);
     TORCH_INTERNAL_ASSERT(list.isList());
     return list;
+  } else if (
+      type->cast<ListType>() &&
```

Comment:
Could you please elaborate why this change is needed? Also, this touches one of the foundational files, so it's better to separate this change from everything else and test separately. cc @eellison too, he should know this part better

Reply:
I'm actually not sure about "how needed" this is - for driving this functionality from Python, it's not. However, if I try to parse the generated IR from C++, with `parseIR`, it hits this branch. In other words, looks like we can't parse what we serialize.

I don't know what the best resolution would be here. I'm worried about the fidelity of our IR parser if it creates types we later cannot recognize. I'm also concerned about the fact that we have different ways to represent a list of constant integers. While I believe it's outside the scope of this PR to solve these issues, I think we should fix them in a more principled way.

Reply:
Ah, I see. I think I've seen a similar problem before - it's a place in the JIT IR Parser that got slightly stale (originally we used the `is` for lists of ints, then changed to `ival`, but forgot to update the parser). A proper way would probably be to fix the IR parser - I could provide you more details if you're interested in doing that, but I don't know if you'd like to diverge to that work. If not, it's probably better to avoid this change anyway and just try not to hit it in the tests.

Reply:
Done, reverted and we'll wait on https://github.com/pytorch/pytorch/pull/43785 to land.